crates/red_knot_python_semantic/src/types/unpacker.rs

```diff
@@ -72,11 +72,9 @@ impl<'db> Unpacker<'db> {
         value_ty: Type<'db>,
     ) {
         match target {
-            ast::Expr::Name(target_name) => {
-                self.targets.insert(
-                    target_name.scoped_expression_id(self.db(), self.scope),
-                    value_ty,
-                );
+            ast::Expr::Name(_) | ast::Expr::Attribute(_) => {
+                self.targets
+                    .insert(target.scoped_expression_id(self.db(), self.scope), value_ty);
```

Comment:
Can we add test cases when the unpacking target is a starred attribute? Like:
```py
In [9]: class Foo:
   ...:     def __init__(self):
   ...:         self.x = (1, 2)
   ...: 
   ...: 
   ...: foo = Foo()
   ...: print(foo.x)
   ...: 
   ...: (a, *foo.x) = (1, 2, 3)
   ...: print(foo.x)
(1, 2)
[2, 3]
```

Ignore if it's already present but I didn't find it in the diff of this PR.

Reply:
(We don't yet support inferring the type here because it requires generic support over list)